canonicalwebteam/blog/common_view_logic.py

```diff
@@ -5,39 +5,145 @@
 group_cache = {}
 
 
+class BlogViews:
+    def __init__(self, tag_ids, excluded_tags, blog_title, tag_name):
+        self.tag_ids = tag_ids
+        self.excluded_tags = excluded_tags
+        self.blog_title = blog_title
+        self.tag_name = tag_name
+
+    def get_index(self, page=1, category="", enable_upcoming=True):
+        category_id = ""
+        if category:
+            category_resolved = api.get_category_by_slug(category)
+            category_id = category_resolved["id"]
+
+        upcoming = []
+        featured_articles = []
+        featured_article_ids = []
```

Comment:
This line could be removed. If we move the following outside the `IF`, just before we do the get_articles call.
```
            featured_article_ids = [
                article["id"] for article in featured_articles
            ]
```

Reply:
liked it, coded it, pushed it